### PR TITLE
EXAMPLES.md corrections

### PR DESCRIPTION
**Pull request by JasonPoll**
While reviewing and testing @mattbrictson 's #524, I found myself referencing `EXAMPLES.md` and noted some inaccuracies. 

_Hoping it's ok to open this PR without a corresponding issue discussing this first._

### CODE REVIEW

File: EXAMPLES.md [hunk 1]
```diff
@@ -148,9 +145,6 @@ end
 This spares one from having to figure out the correct escaping sequences for
 something like "echo(:cat, '...?...', '> /etc/sudoers.d/yolo')".
 
-**Note:** The `upload!()` method doesn't honor the values of `within()`, `as()`
-etc, this will be improved as the library matures, but we're not there yet.
```
Comment (JasonPoll) [inline EXAMPLES.md:152]: This (and the above) appeared to be inaccurate based on [my testing of #524](https://github.com/capistrano/sshkit/pull/524#pullrequestreview-1799265301).

File: EXAMPLES.md [hunk 2]
```diff
@@ -235,16 +230,16 @@ end
 
 ```ruby
 # The default format is pretty, which outputs colored text
-SSHKit.config.format = :pretty
+SSHKit.config.use_format :pretty
```
Comment (JasonPoll) [inline EXAMPLES.md:233]: These appeared to have never been updated in 2cb6326.

File: EXAMPLES.md [hunk 3]
```diff
@@ -254,7 +249,7 @@ module SSHKit
   module Formatter
     class MyFormatter < SSHKit::Formatter::Abstract
       def write(obj)
-        case obj.is_a? SSHKit::Command
+        if obj.is_a? SSHKit::Command
```
Comment (JasonPoll) [inline EXAMPLES.md:252]: At least with Ruby 3.2.2, this `case` produced an error:

```
SyntaxError: unexpected `end', expecting `when'
```

File: EXAMPLES.md [hunk 4]
```diff
@@ -160,7 +154,8 @@ end
 ```
 
 In this case the `recursive: true` option mirrors the same options which are
-available to [`Net::{SCP,SFTP}`](http://net-ssh.github.io/net-scp/).
+available to [`Net::SCP`](http://net-ssh.github.io/net-scp/) and
```
Comment (mattbrictson) [inline EXAMPLES.md:157]: @JasonPoll can you change the SCP link to https://github.com/net-ssh/net-scp ? It looks like http://net-ssh.github.io/net-scp/  is out of date (the header says "Last Update: 2014").

File: EXAMPLES.md (hunk 3 above)
Comment (mattbrictson) [reply EXAMPLES.md:252]: Good catch!

**Review by mattbrictson — CHANGES_REQUESTED** [review]
Thanks for this! One small change requested; see comments

File: EXAMPLES.md (hunk 4 above)
Comment (JasonPoll) [reply EXAMPLES.md:157]: Can do.

**Review by mattbrictson — APPROVED** [review]

**Pull request merged**